crates/sui-framework/sources/balance.move

```diff
@@ -27,15 +31,39 @@ module sui::balance {
         self.value
     }
 
+    /// Get the `Supply` value.
+    public fun supply_value<T>(supply: &Supply<T>): u64 {
+        supply.value
+    }
+
+    /// Create a new supply for type T.
+    public fun create_supply<T: drop>(_witness: T): Supply<T> {
```

Comment:
Exposing this publicly rather than as a `friend` to `Coin` makes me a bit nervous. I see that we need to do so for `Sui::new()`, but is it possible to avoid that by having genesis continue to store the `TreasuryCap<SUI>`?

Reply:
But it's exactly the same thing as we do in Coin. This unblocks custom Coins to be created such as `RegulatedCoin`.

Reply:
Ah, I might have misunderstood the `RegulatedCoin` use-case. If `Supply` is intended to be create-able outside of `Coin` in that use-case, then I have no problem with the code as-is.

I only wanted the friend restriction if there was no reason to use `Supply` outside of `Coin` except in Sui genesis.

Reply:
Yeah. RegulatedCoin is now forced to use Coin just to create currency. It is also updated in this PR, so that RC is now free of Coin.